src/particle_header.F90

```diff
@@ -141,6 +186,60 @@ subroutine initialize_particle(this)
   end subroutine initialize_particle
 
 !===============================================================================
+! PARTICLE_TO_BUFFER copies particle information to the ParticleBuffer data type
+! for sending via MPI
+!===============================================================================
+
+  subroutine particle_to_buffer(part, buf)
+
+    type(Particle),       intent(in)  :: part
+    type(ParticleBuffer), intent(out) :: buf
+
+    buf % id              = part % id
+    buf % type            = part % type
+    buf % wgt             = part % wgt
+    buf % E               = part % E
+    buf % last_xyz        = part % last_xyz
+    buf % surface         = part % surface
+    buf % cell_born       = part % cell_born
+    buf % n_collision     = part % n_collision
+    buf % stored_xyz      = part % stored_xyz
+    buf % stored_uvw      = part % stored_uvw
+    buf % prn_seed        = part % prn_seed
+    buf % stored_distance = part % stored_distance
+    buf % fly_dd_distance = part % fly_dd_distance
+
+  end subroutine particle_to_buffer
+
+!===============================================================================
+! BUFFER_TO_PARTICLE copies particle information out of ParticleBuffer type
+! received via MPI back into a particle data type
+!===============================================================================
+
+  subroutine buffer_to_particle(buf, part)
+
+    type(ParticleBuffer), intent(in)  :: buf
+    type(Particle),       intent(out) :: part
+
+    part % id              = buf % id
```

Comment:
Add a `call part % initialize()` at the start of this routine so we can be sure that there is no left-over state from the last particle.